paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -50,7 +50,7 @@ import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinAndroidTarget
 import java.util.Locale
 
 @Suppress("unused")
-class PaparazziPlugin : Plugin<Project> {
+internal class PaparazziPlugin : Plugin<Project> {
```

Comment:
@JakeWharton is there any reason why Paraphrase keeps the plugin type public? https://github.com/cashapp/paraphrase/blob/main/plugin/api/plugin.api

Seems like Gradle is able to access the plugin and task types without the need to keep them plugin, which I guess makes sense, since build stuff should mostly be implementation detail.

Reply:
I suppose this works because `internal` classes are public from the JVM's point of view, but I think Gradle plugins are semantically public in the sense that the external Gradle tool needs to instantiate them and invoke their API.

Reply:
Also `project.pluginManager.apply(PaparazziPlugin::class.java)` is a way of applying, I would suspect this breaks some users. Migration is easy though, so you can add to release notes to use `.apply("app.cash.paparazzi")` instead.

Reply:
ah, fair point.  although I prefer the string-based id approach, I'm fine with leaving this public.

Reply:
Yes. Public for apply and withType and such. Plus Gradle requires it (at least in bytecode).